### PR TITLE
fix: mention and community membership issues

### PR DESCRIPTION
**Pull request by NaingAmity**

### CODE REVIEW

File: src/hooks/useSearch.ts [hunk 1]
```diff
@@ -1,14 +1,50 @@
...
+          search: text,
+          limit: 5,
+          sortBy: 'firstCreated',
+          memberships: ['member'],
+        },
+        ({ data, error, hasNextPage, onNextPage }) => {
+          if (error) return null;
+          hasNextPage
+            ? (onNextPageRef.current = onNextPage)
+            : (onNextPageRef.current = null);
```
Comment (frankent) [inline src/hooks/useSearch.ts:32]: Can we use this instead ?

```suggestion
          onNextPageRef.current = hasNextPage ? onNextPage : null;
```

File: src/redux/slices/uiSlice.ts [hunk 2]
```diff
@@ -64,6 +68,7 @@ const uiSlice = createSlice({
       state.targetName = '';
       state.postSetting = null;
       state.needApprovalOnPostCreation = true;
+      state.isPublic = true;
```
Comment (frankent) [inline src/redux/slices/uiSlice.ts:71]: Should we force this as always TRUE ?

File: src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx [hunk 3]
```diff
@@ -1,62 +1,83 @@
...
+    const unsubscribe = CommunityRepository.Membership.getMembers(
       option,
-      ({ data: members, onNextPage, hasNextPage, loading: fetching }) => {
+      ({ data, onNextPage, hasNextPage, loading: fetching, error }) => {
         setLoading(fetching);
+        if (error) return setRefreshing(false);
         if (!fetching) {
-          setMemberList([...members]);
-          const userArray: UserInterface[] = members.map((member) => {
+          setMemberList([...data]);
```
Comment (frankent) [inline src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx:59]: Why not just put `data` to `setMemberList` directly ?

File: src/screens/PostDetail/index.tsx [hunk 4]
```diff
@@ -291,107 +286,6 @@ const PostDetail = () => {
     console.log('post:', post);
```
Comment (frankent) [inline src/screens/PostDetail/index.tsx:286]: Can we remove this line ?

**Review by frankent — CHANGES_REQUESTED** [review]
Please check my comment

File: src/hooks/useSearch.ts (hunk 1 above)
Comment (NaingAmity) [reply src/hooks/useSearch.ts:32]: Done krub

File: src/redux/slices/uiSlice.ts (hunk 2 above)
Comment (NaingAmity) [reply src/redux/slices/uiSlice.ts:71]: Changed default to false

File: src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx (hunk 3 above)
Comment (NaingAmity) [reply src/screens/CommunityMemberDetail/Components/CommunityMembersTab.tsx:59]: We are subscribing members from community and we need a new array for every members changed in that array. 
So I make data array as immutable to rerender the list component

File: src/screens/PostDetail/index.tsx (hunk 4 above)
Comment (NaingAmity) [reply src/screens/PostDetail/index.tsx:286]: done

**Review by frankent — APPROVED** [review]
LGTM

**Pull request merged**